dataprofiler/profilers/profile_builder.py

```diff
@@ -1402,18 +1477,26 @@ def tqdm(l):
         # Newly introduced features (python3.8) improves the situation
         sample_ids = np.array(sample_ids)
 
-        # Create structured profile objects
-        new_cols = set()
-        for col in data.columns:
-            if col not in self._profile:
-                self._profile[col] = StructuredColProfiler(
+        # Create StructuredColProfilers upon initialization
+        # Record correlation between columns in data and index in _profile
+        new_cols = False
+        if not initialized:
```

Comment:
since this is really the only place needed, can we just check the len of the profile here instead of using a var?

Reply:
Implemented